.github/workflows/iroha2-dev-pr.yml

```diff
@@ -137,3 +137,45 @@ jobs:
           file: Dockerfile
           # This context specification is required
           context: .
+
+  client-cli-test:
+    runs-on: ubuntu-latest
+    container:
+      image: alexstrokelive/iroha2-ci:pytest-nightly-2023-06-25
+    timeout-minutes: 60
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build iroha_client_cli
+        run: |
+          cargo build --bin iroha_client_cli --release
+          cargo build --bin kagami --release
+          cargo build --bin iroha --release
+
+      - name: Mark binaries as executable
+        run: |
+          chmod +x $CLIENT_CLI_DIR/
+
+      - name: Run Iroha 2 on bare metal
+        run: |
+          ./scripts/test_env.py setup   
+
+      - name: Copy Iroha 2 client config
+        run: |
+          cp ./configs/client/config.json $CLIENT_CLI_DIR/
+
+      - name: Set up .env for client_cli tests
+        run: |
+          echo "CLIENT_CLI_DIR=$CLIENT_CLI_DIR" >> ./client_cli/pytests/.env
+          echo "TORII_API_PORT_MIN=8080" >> ./client_cli/pytests/.env
+          echo "TORII_API_PORT_MAX=8083" >> ./client_cli/pytests/.env
+
+      - name: Install dependencies using Poetry
+        working-directory: client_cli/pytests
+        run: |
+          poetry install
```

Comment:
should we do this in the `Dockerbuild` as well?

Reply:
it should speed up the process

Reply:
I took a closer look and I don't see that as a possibility.

Reply:
We might think how to cache some dependencies during these tests execution to speed up the whole process in further

Reply:
> I took a closer look and I don't see that as a possibility.

but why not?

Reply:
> > I took a closer look and I don't see that as a possibility.
> 
> but why not?

To `poetry install` we need the last `pyproject.toml` and `poetry.lock` files which we get using fresh checkout in actions and which we don't have at the image.

Reply:
That's why I said you should do it in both places. In the image and then later in the workflow job. That way the 2nd call to `python poetry` will have less work to do